### ISSUE TITLE
UmbracoContext can not retrieve IPublishedContent in a UmbracoAuthorizedJsonController

### ISSUE THREAD

**Issue by Lachanophobia** [umbraco/Umbraco-CMS#10924]
### Which *exact* Umbraco version are you using? For example: 8.13.1 - don't just write v8

V9 RC2

### Bug summary

I have a controller which inherits from the UmbracoAuthorizedJsonController and it is used by a plugin. It needs to retrieve a saved value from a "settings" node and return it to Backoffice.

In the UmbracoAuthorizedJsonController I don't have access to "Umbraco.Content()" so I am trying to get the umbraco context with the following 2 ways. 

Both seem to bring back the context but when I am trying to get a node (any node), it returns always null.
 
 Both ways are are tested and working fine in a cshtml view also it was working fine in the U9 RC1

### Specifics

Here is the code I am using to retrieve the context a the node.
Both if statements are "true" but root and root2 always return null.

```
using System;
using System.Linq;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.DependencyInjection;
using Umbraco.Cms.Core.Web;
using Umbraco.Cms.Web.BackOffice.Controllers;
using Umbraco.Cms.Web.Common;
using Umbraco.Cms.Web.Common.Attributes;

namespace Eshop.Web.BusinessLogic.Controllers.UmbracoAuthorizedControllers
{
    [IsBackOffice]
    public class SeoPageNameController : UmbracoAuthorizedJsonController
    {
        private readonly IHttpContextAccessor _contextAccessor;
        private readonly IUmbracoContextAccessor _umbracoContextAccessor;
        private UmbracoHelper _umbracoHelper;

        public SeoPageNameController(IHttpContextAccessor contextAccessor,
            IUmbracoContextAccessor umbracoContextAccessor)
        {
            _contextAccessor = contextAccessor;
            _umbracoContextAccessor = umbracoContextAccessor;
        }

        [HttpGet]
        public string GetDefault()
        {
            var hasContext = _umbracoContextAccessor.TryGetUmbracoContext(out var umbracoContext);

            if (hasContext)
            {
                var root = umbracoContext.Content.GetAtRoot().FirstOrDefault();
            }

            
            if (_contextAccessor.HttpContext == null) throw new NullReferenceException("Can not get umbraco context");
            _umbracoHelper = _contextAccessor.HttpContext.RequestServices.GetRequiredService<UmbracoHelper>();
            var root2 = _umbracoHelper.ContentAtRoot().FirstOrDefault();


            return "";
        }
    }
}
```

### Steps to reproduce

- Create a node in the Umbraco BackOffice
- Create a UmbracoAuthorizedJsonController like the example above
- Create a Umbraco Plugin with a basic angular controller just to call the UmbracoAuthorizedJsonController 
- Try to retrieve a node 

### Expected result / actual result

Retrieve the IPublishedContent node from the Umbraco Context in a UmbracoAuthorizedJsonController

**Comment by bergmania** [comment]
Hi @Lachanophobia, thanks to reporting.. 

I can't reproduce it on my first attempt. Both if's return the expected root item:

![image](https://user-images.githubusercontent.com/1561480/130401028-ef02dc2d-eafe-496b-aa8c-3e954ee97c9f.png)

Are you sure the item in root is published?

Do you get any results if you are using 
``` 
var root = umbracoContext.Content.GetAtRoot(preview:true).FirstOrDefault();
```
or specify the culture (replace "dk" with your existing cultures)
``` 
var root = umbracoContext.Content.GetAtRoot(culture:"da").FirstOrDefault();
```

**Comment by Lachanophobia** [comment]
Hi @bergmania,

Yes my nodes are published and it worked with a specified culture! Now that I am thinking it again, its not bug? I mean, Umbraco probably doesn't know which culture I am browsing in the backoffice so in my UmbracoAuthorizedJsonController doesn't know which node to pick..

Could be that the answer? What do you thing?

**Comment by bergmania** [comment]
Yep, that makes sense.. If the content is variant, it should be requested by the variant. 
Otherwise it requests for the invariant culture (culture = ""), and it makes sense if it do not find anything there.

**Comment by Lachanophobia** [comment]
Thank you  @bergmania !
